core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -173,26 +174,34 @@ private void refreshContexts() {
         try {
             synchronized (contexts) {
                 final List<Endpoint> newSelectedEndpoints = healthCheckStrategy.getSelectedEndpoints();
+                final Set<Endpoint> newSelectedEndpointsSet = new HashSet<>(newSelectedEndpoints);
```

Comment:
Could remove this original local variable to avoid unwanted access?
```java 
final Set<Endpoint> newSelectedEndpoints = new HashSet<>(healthCheckStrategy.getSelectedEndpoints());
```

Reply:
That's a good suggestion.